packages/storybook/.storybook/TOC.tsx

```diff
@@ -0,0 +1,61 @@
+import { mergeStyles } from '@fluentui/react';
+import React, { useState } from 'react';
+import { BackToTop, TableOfContents } from 'storybook-docs-toc';
+
+/**
+ * Table Of Contents component used for displaying a floating table of content
+ * inside Storybook Docs.
+ * @param props 
+ * @returns 
+ */
+export const TOC = (props: {children: React.ReactNode}): JSX.Element => {
+
+  const [narrowDocs, setnarrowDocs] = useState(false);
+  const [hideToc, setHideToc] = useState(false);
+
+  window.matchMedia('(max-width: 1200px)').addEventListener('change', (e) => {
```

Comment:
Could we put these inside a useEffect and when the component unmounts unregister the listeners?

Reply:
Yes. Thanks for pointing it out.